tools/build_with_debinfo.py

```diff
@@ -78,6 +78,9 @@ def create_build_plan() -> list[tuple[str, str]]:
         if line.startswith(": &&") and line.endswith("&& :"):
             line = line[4:-4]
         line = line.replace("-O2", "-g").replace("-O3", "-g")
+        # Build Metal shaders with debug infomation
+        if "xcrun metal " in line and "-frecord-sources" not in line:
+            line += " -frecord-sources -gline-tables-only"
```

Comment:
Thanks for this. I wonder why we're passing `-gline-tables-only` rather than `-g0` or `-g1` -- I assume we don't care about frame variables and other info for now (in particular for profiling).